packages/core/src/signer/json-rpc-account.ts

```diff
@@ -0,0 +1,40 @@
+import { getAddress, type ByteArray, type Hex, type WalletClient } from "viem";
+import type { SmartAccountSigner } from "./types";
+import type { SignTypedDataParams } from "../account/types";
+
+export class JsonRpcAccountSigner implements SmartAccountSigner {
+  private client: WalletClient;
+
+  constructor(client: WalletClient) {
+    this.client = client;
+  }
+
+  getAddress: () => Promise<`0x${string}`> = async () => {
+    let addresses = await this.client.getAddresses();
+    return getAddress(addresses[0]);
+  };
+
+  readonly signMessage: (
+    message: string | Hex | ByteArray
+  ) => Promise<`0x${string}`> = async (message) => {
+    if (typeof message === "string") {
```

Comment:
`Hex` is a string too, but needs to be passed as a `raw` parameter below

```suggestion
    if (typeof message === "string" && !message.startsWith("0x")) {
```